tuple/xxhash.mbt

```diff
@@ -0,0 +1,192 @@
+// Copyright 2024 International Digital Economy Academy
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// https://github.com/Cyan4973/xxHash/blob/dev/doc/xxhash_spec.md#xxh32-algorithm-description
+
+let gPRIME1 = 0x9E3779B1U
+
+let gPRIME2 = 0x85EBCA77U
+
+let gPRIME3 = 0xC2B2AE3DU
+
+let gPRIME4 = 0x27D4EB2FU
+
+let gPRIME5 = 0x165667B1U
+
+pub fn hash[T0 : Hash, T1 : Hash](self : (T0, T1)) -> Int {
```

Comment:
we are thinking of change the Hash trait into something like
```
trait Hash {
   hash_fold(Self, UInt) -> UInt
   hash(Self) -> Int
}
```
thoughts?

Reply:
or maybe 
```
trait Hash {
  hash(Self, ~st : UInt = 0) -> UInt
}
```

Reply:
So calculating the combination of hashes is making a recursive call to the hash function? That makes sense.

Reply:
I think it makes sense. Or we could provide a hash combinator, like the [.NET HashCode class](https://github.com/dotnet/runtime/blob/1d1bf92fcf43aa6981804dc53c5174445069c9e4/src/libraries/System.Private.CoreLib/src/System/HashCode.cs)